src/AVFoundation/Events.cs

```diff
@@ -69,30 +71,30 @@ public InternalAVAudioPlayerDelegate ()
 		[Preserve (Conditional = true)]
 		public override void FinishedPlaying (AVAudioPlayer player, bool flag)
 		{
-			if (cbFinishedPlaying != null)
+			if (cbFinishedPlaying is not null)
 				cbFinishedPlaying (player, new AVStatusEventArgs (flag));
 			if (player.Handle == IntPtr.Zero)
 				throw new ObjectDisposedException ("player", "the player object was Dispose()d during the callback, this has corrupted the state of the program");
 		}
 	
 		[Preserve (Conditional = true)]
-		public override void DecoderError (AVAudioPlayer player, NSError  error)
+		public override void DecoderError (AVAudioPlayer player, NSError?  error)
 		{
-			if (cbDecoderError != null)
+			if (cbDecoderError is not null && error is not null)
 				cbDecoderError (player, new AVErrorEventArgs (error));
```

Comment:
I think this is better, that way we won't swallow errors, but let the app developer handle them even if we're not told what's going wrong:

```suggestion
			if (cbDecoderError is not null)
				cbDecoderError (player, error is not null ? new AVErrorEventArgs (error) : null);
```